From eff97ad5f80ab7f00d5ff67ee1473e55962d3f44 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 04 Jan 2021 19:31:39 +0200
Subject: [PATCH] [SW-0] preboot: pll: introduce dev spec pll init

Introduce device specific PLLs initialization function (callback).
This is in preparation for adding back the PLLs initialization functions
as needed by Gaudi HBM init code.
Since the new function is introduced, also move the Goya2 PLDM PLLs init
call to the new function.

Change-Id: I47aab460269abcdef774eb7936777892c85ea0d7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.h b/common/pll.h
index 971963d..6cca608 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -41,6 +41,7 @@
 	int     div_sel[HL_PLL_DIV_SIZE];
 };
 
+void hl_dev_pll_init(void);
 void hl_pll_init(struct hl_pll_cfg *hl_pll_cfg, int size);
 void udelay_init(void);
 u64 hl_pll_timestamp_freq_get(void);
diff --git a/goya2_arc_gnu/pldm.c b/goya2_arc_gnu/pldm.c
index 1acf7a6..bb8a392 100644
--- a/goya2_arc_gnu/pldm.c
+++ b/goya2_arc_gnu/pldm.c
@@ -53,10 +53,14 @@
 	return SYS_CLOCK_FREQUENCY;
 }
 
-void hl_uart_init(void)
+void hl_dev_pll_init(void)
 {
 	/* Bump up UART frequency */
 	hl_pll_init(goya2_pll_cfg, 1);
+}
+
+void hl_uart_init(void)
+{
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
diff --git a/preboot/preboot.c b/preboot/preboot.c
index bd8f8e8..063ae2c 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -246,6 +246,7 @@
 		      sizeof(preboot_en_caps));
 }
 
+__weak void hl_dev_pll_init(void) {}
 __weak void hl_clock_gating_init(void) {}
 __weak void preboot_conf_iatu_access(bool enable) {}
 void preboot_main(void)
@@ -265,6 +266,8 @@
 	 */
 	udelay_init();
 
+	hl_dev_pll_init();
+
 	hl_uart_init();
 
 	preboot_sp_sram_init();
